### PR TITLE
remove unwrapAmount and check WETH amount only

### PR DESCRIPTION
**Pull request by charlesjhongc**

### CODE REVIEW

File: contracts/LimitOrderSwap.sol [hunk 1]
```diff
@@ -92,6 +92,10 @@ contract LimitOrderSwap is ILimitOrderSwap, Ownable, TokenCollector, EIP712 {
                 if (makingAmount > orderAvailableAmount) revert NotEnoughForFill();
                 takerTokenAmounts[i] = ((makingAmount * order.takerTokenAmount) / order.makerTokenAmount);
 
+                if (order.takerToken == address(weth)) {
+                    wethToPay += takerTokenAmounts[i];
+                }
```
Comment (pilagod) [inline contracts/LimitOrderSwap.sol:97]: I think these block could be placed after zero amount check (line 99)

File: test/forkMainnet/LimitOrderSwap/GroupFill.t.sol [hunk 2]
```diff
@@ -233,7 +215,8 @@ contract GroupFillTest is LimitOrderSwapTest {
         // scenario
         // maker tokens (input): 5 WETH, 8000 USDT
         // taker tokens (output): 1 ETH, 3 WETH, 8000 USDT
-        // profit for arbitrageur: 1 ETH/WETH
+        // should unwrap 4 ether and leave 1 WETH
```
Comment (pilagod) [inline test/forkMainnet/LimitOrderSwap/GroupFill.t.sol:218]: I think this case unwraps only 2 WETH (5 WETH - 3 WETH), and 1 ETH for order and another 1 ETH for arbitrageur's profit, right?

Comment (charlesjhongc) [reply test/forkMainnet/LimitOrderSwap/GroupFill.t.sol:218]: Yeah, you're right. Unwrap 3 and profit is 1.

Comment (charlesjhongc) [reply test/forkMainnet/LimitOrderSwap/GroupFill.t.sol]: Fixed in 0708ce1

File: contracts/LimitOrderSwap.sol (hunk 1 above)
Comment (charlesjhongc) [reply contracts/LimitOrderSwap.sol:99]: Fixed in 0708ce1

**Review by NIC619 — APPROVED** [review]

**Review by pilagod — APPROVED** [review]

**Pull request merged**